### PR TITLE
dev branch import fix

### PR DESCRIPTION
**Pull request by Abrams27**
well, dev branch finaly works!

things I changed:
- now almost every call uses `--bes_backend...` flag
- `.waitFor` is *not* called every time - probably the main reason of timeouts, previously streams was read without waiting, as it is now
- added more logging - .bazelbsp.log` now contains more info - every single call, params and responses

### CODE REVIEW

**Review by andrefmrocha — CHANGES_REQUESTED** [review]
Overall, it looks great. I love the further logging that you added! Just have two regarding the code itself.

Another bigger question I have is that this doesn't really say much as to why the current `dev` branch could import a project like the sample-repo but not this one. Any clue as to why that is the case? I think that could be important in order to guarantee we have tests for that now.

Going to also look into testing this locally and see how it's working

File: src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java [hunk 1]
```diff
@@ -79,11 +79,11 @@ public BazelRunnerBuilder withKinds(List<BazelQueryKindParameters> bazelParamete
     return this;
   }
 
-  public BazelProcessResult executeBazelCommand() {
+  public BazelProcess executeBazelCommand() {
```
[the remark this answers is not in the document]
Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java:82]: well, this method is used at the very beginning of the server, even before we know the bes server port

[the remark this answers is not in the document]
Comment (andrefmrocha) [reply src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java:82]: But I'm not sure there's any real reason as to why that should be done in that order

**Comment by andrefmrocha** [comment]
It would seem there a lot of empty lines being outputted as logs, leading to weird empty spaces in the logs. Should we try and address that here or should we leave that for a later PR?

**Comment by andrefmrocha** [comment]
> It would seem there a lot of empty lines being outputted as logs, leading to weird empty spaces in the logs. Should we try and address that here or should we leave that for a later PR?

This same question can be applied to the build request.

File: src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java:82]: well, it's the consequence of this circular initialization, nevertheless i'll try to change it

**Comment by Abrams27** [comment]
> It would seem there a lot of empty lines being outputted as logs, leading to weird empty spaces in the logs. Should we try and address that here or should we leave that for a later PR?

as I remember correctly it happens also at main branch, so I would prefer to leave it for later PR


> > It would seem there a lot of empty lines being outputted as logs, leading to weird empty spaces in the logs. Should we try and address that here or should we leave that for a later PR?
> 
> This same question can be applied to the build request, since it's also failing

the same here, further investigation for later

File: src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/server/bazel/BazelRunnerBuilder.java:82]: after little research, it will make entire proces even more ambiguous, bo maybe I'll leave it as it is now. I think that there is nothing bad in those 2 separate methods

File: src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
   <Appenders>
-
     <Console name="Console" target="SYSTEM_OUT">
       <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
     </Console>
 
-    <File append="false" fileName="all.log" immediateFlush="false" name="File">
-      <PatternLayout pattern="%d{yyy-MM-dd HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+    <File name="File" fileName=".bazelbsp/bazelbsp.log" append="false">
+      <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
```
Comment (agluszak) [inline src/main/resources/log4j2.xml:10]: why did you remove the date?

Comment (Abrams27) [reply src/main/resources/log4j2.xml:10]: dont know, will rollback

**Comment by Abrams27** [comment]
> Overall, it looks great. I love the further logging that you added! Just have two questions regarding the code itself.
> 
> Another bigger question I have is that this doesn't really say much as to why the current `dev` branch could import a project like the sample-repo but not this one. Any clue as to why that is the case? I think that could be important in order to guarantee we have tests for that now.
> 
> Going to also look into testing this locally and see how it's working

Probably, because of size and time required to run bazel commands and somehow `.waitFor` dont block entire flow

**Review by gerardd33 — APPROVED** [review]
Great job! Especially with the logging

File: src/main/java/org/jetbrains/bsp/bazel/server/bazel/data/BazelProcess.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  private void logBazelMessage(BazelProcessResult bazelProcessResult, int exitCode) {
+    if (exitCode == OK_EXIT_CODE) {
+      logBazelMessage(bazelProcessResult);
+    } else {
+      logBazelError(bazelProcessResult);
+    }
+  }
+
+  private void logBazelMessage(BazelProcessResult bazelProcessResult) {
+    String message = bazelProcessResult.getJoinedStderr();
```
Comment (agluszak) [inline src/main/java/org/jetbrains/bsp/bazel/server/bazel/data/BazelProcess.java]: is logging stderr here correct? if yes, maybe a comment clarifying that would be nice

Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/server/bazel/data/BazelProcess.java]: well, it is correct because it's copied, but why I don't know. Probably I will fix it somehow in the following PR trying to remove redundant empty lines in those logs

**Review by andrefmrocha — APPROVED** [review]
:rocket:

**Comment by Abrams27** [comment]
> > Probably, because of the project size and time required to run bazel commands, so _somehow_ `.waitFor` blocked entire flow at `dev`
> 
> This leads back to that question about how it would be great to use our own repo for testing the bsp protocol that @agluszak mentioned at the last meeting.

yea, I'll do it later
